java-checks/src/main/java/org/sonar/java/checks/OptionalAsParameterCheck.java

```diff
@@ -48,11 +48,17 @@ public List<Tree.Kind> nodesToVisit() {
 
   @Override
   public void visitNode(Tree tree) {
-    for (VariableTree parameter : ((MethodTree) tree).parameters()) {
-      TypeTree typeTree = parameter.type();
-      Optional<String> msg = expectedTypeInsteadOfOptional(typeTree.symbolType());
-      if (msg.isPresent()) {
-        reportIssue(typeTree, msg.get());
+    final var methodTree = (MethodTree) tree;
+
+    // If the method is overriding something, the user has no control over the parameter type here, so we should not raise an issue.
+    if (methodTree.isOverriding() == Boolean.FALSE) {
```

Comment:
Interview question: What is the result of this java expression?
```java
(new Boolean(false) == Boolean.FALSE)
```
So here, because we are not `kotlin` developers :stuck_out_tongue: but `java` developers :grin:, we should use:
```java
Boolean.FALSE.equals(methodTree.isOverriding())
```

Except for this, I approve this PR, once fixed you can merge.

Reply:
:eyes: